x-pack/plugins/enterprise_search/public/applications/shared/telemetry/send_telemetry.tsx

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useContext, useEffect } from 'react';
+
+import { HttpHandler } from 'src/core/public';
+import { KibanaContext, IKibanaContext } from '../../index';
+
+interface ISendTelemetryProps {
+  action: 'viewed' | 'error' | 'clicked';
+  value: string; // e.g., 'setup_guide'
+}
+
+interface ISendTelemetry extends ISendTelemetryProps {
+  http(): HttpHandler;
+  product: 'app_search' | 'workplace_search' | 'enterprise_search';
+}
+
+/**
+ * Base function - useful for non-component actions, e.g. clicks
+ */
+
+export const sendTelemetry = ({ http, product, action, metric }: ISendTelemetry) => {
+  return http
+    .put(`/api/${product}/telemetry`, {
+      headers: { 'content-type': 'application/json' },
+      body: JSON.stringify({ action, metric }),
+    })
+    .catch(e => {
+      throw e;
+    });
+};
+
+/**
+ * React component helpers - useful for on-page-load/views
+ * TODO: SendWorkplaceSearchTelemetry and SendEnterpriseSearchTelemetry
+ */
+
+export const SendAppSearchTelemetry: React.FC<ISendTelemetryProps> = ({ action, metric }) => {
+  const { http } = useContext(KibanaContext) as IKibanaContext;
+
+  useEffect(() => {
+    sendTelemetry({ http, action, metric, product: 'app_search' });
+  }, [action, metric, http]);
```

Comment:
Is there any valid use case where `action`, `metric`, or `http` would actually change and we'd want to update the metrics? Just wondering if this should be `[]` instead.

Reply:
So this is an eslint thing, personally I do think it should be `[]` but when I set it to that eslint's `react-hooks/exhaustive-deps` rule flags an error: https://github.com/facebook/react/issues/14920

There's a ton of discourse on it on the internet and I'm not a huge fan of the rule, but since it's Kibana's codebase we either play by their rules or put it in a lot of one-liner eslint disables 🤷 

There's actually another TODO comment in engine_overview.tsx where I did end up disabling the line - I think there's another link in that comment that has more context/info.

Reply:
Gotcha. I mean i think it will be fine as is, I'm not concerned about it.

Reply:
Haha yea same, but I also personally wish it did not be that way 😁